### ISSUE TITLE
/* GraphQL */ template strings are not formatted when TypeScript parser is used

### ISSUE THREAD

**Issue by skevy** [prettier/prettier#4679]
**Prettier 1.13.5**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEBHArnAJwE9MBeTAegCpMBxAgQwAcALARQBlMqLMADADpRMI0cCGjMuNIQAUogJYATTElEBWEQEoR44ZMwAzBQQwA5BgFs4EiQYA2Dc1Zv6RAXzueofANwgAGhAIJhgFdGRQBgICCAB3AAVohDRkEAYANwhlQJAAI0YwAGs4GABlJgYwBSgAc2QYAnwgmpkCGATGWssGZEMGexkggCs0AA8AIUKS8pcOGrg+gaGQStNCNJgiJjg0MAIFUNymA9gAdWUYFmQADgAGIJOIGTPGJjST3cIMxaCCODwJjgnQY3V6SH6gzgQRklgUDSa0JAaBqtXscDYuAg8CWUKCMAYeQuSiuyAATPjGAp7KiAMIQSw9NJQaC-EDSOAAFUJqQhyzg7ncQA)
```sh
# Options (if any): 
--parser typescript
```

**Input:**
```jsx
const query = /* GraphQL */ `
      {
    user(   id :   5  )  {
      firstName
 
      lastName
    }
  }
`;
```

**Output:**
```jsx
const query = /* GraphQL */ `
      {
    user(   id :   5  )  {
      firstName
 
      lastName
    }
  }
`;
```

**Expected behavior:**

I expect the GraphQL to be formatted as of Prettier 1.13.

I investigated why this was happening. There is no "leadingComments" property on the AST node when using the TypeScript parser, and thus it doesn't find the `/* GraphQL */` comment.

The `isGraphQL` method should check the `node.comments` array when using the TypeScript parser.

**Comment by skevy** [comment]
Related to https://github.com/prettier/prettier/pull/4395.

cc @tjallingt

**Comment by armano2** [comment]
@skevy @lydell issue is related to `CommentBlock` 
`@typescript-eslint/typescript-estree` provides node with type `Block`

| parser | multiline| single line | 
| -------------- | ---- | ---- |
| babel |  `CommentBlock` |  `CommentLine`
| babel-eslint |  `Block` |  `Line`
| typescript-estree |  `Block` |  `Line`
| espree |  `Block` |  `Line`
| esprima |  `Block` |  `Line`
| flow |  `Block` |  `Line`

**Comment by thorn0** [comment]
Not sure when this was fixed, but it works in 1.18.2.

**Prettier 1.18.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEBHArnAJwE9MBeTAegCpMBxAgQwAcALARQBlMqLMADADpRMI0cCGjMuNIQAUogJYATTElEBWEQEoR44ZMwAzBQQwA5BgFs4EiQYA2Dc1Zv6RAXzueofANwgAGhAIJhgFdGRQBgICCAB3AAVohDRkEAYANwhlQJAAI0YwAGs4GABlJgYwBSgAc2QYAnwgmpkCGATGWssGZEMGexkggCs0AA8AIUKS8pcOGrg+gaGQStNCNJgiJjg0MAIFUNymA9gAdWUYFmQADgAGIJOIGTPGJjST3cIMxaCCODwJjgnQY3V6SH6gzgQRklgUDSa0JAaBqtXscDYuAg8CWUKCMAYeQuSiuyAATPjGAp7KiAMIQSw9NJQaC-EDSOAAFUJqQhyzg7ncQA)
```sh
--parser typescript
```

**Input:**
```tsx
const query = /* GraphQL */ `
      {
    user(   id :   5  )  {
      firstName
 
      lastName
    }
  }
`;
```

**Output:**
```tsx
const query = /* GraphQL */ `
  {
    user(id: 5) {
      firstName

      lastName
    }
  }
`;

```
